### PR TITLE
refactor: remove unused escape function

### PR DESCRIPTION
**Pull request by rvantonder**
See inline comment.

### CODE REVIEW

File: internal/search/query/syntax/parser.go [hunk 1]
```diff
@@ -49,19 +46,6 @@ func Parse(input string) (ParseTree, error) {
 	return exprs, nil
 }
 
-// MaybeEscapeValue will escape value if it needs escaping.
-func MaybeEscapeValue(value string) string {
```
Comment (rvantonder) [inline internal/search/query/syntax/parser.go]: This function is never applied to any search input. It is only used in tests... for something that we don't use. So it goes.

File: cmd/frontend/graphqlbackend/search_test.go [hunk 2]
```diff
@@ -230,11 +229,11 @@ func testStringResult(result *searchSuggestionResolver) string {
 	var name string
 	switch r := result.result.(type) {
 	case *RepositoryResolver:
-		name = "repo:" + syntax.MaybeEscapeValue(string(r.repo.Name))
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/search_test.go]: weird, how did this end up not used? We used to use it for generating filter suggestions. Maybe a refactor broke that?

**Comment by keegancsmith** [comment]
This was added to fix a bug, but I think I did it ineffectively! The original PR forgot to use it outside of the tests! https://github.com/sourcegraph/sourcegraph/pull/7031

Rather than revert this, we need to actually use this function in the place we generate these filter suggestions.

**Comment by keegancsmith** [comment]
It seems the code which generates these suggestions is quite different. This PR you can probably merge, and we need to do the escaping in the frontend layer?

File: cmd/frontend/graphqlbackend/search_test.go (hunk 2 above)
Comment (rvantonder) [reply cmd/frontend/graphqlbackend/search_test.go]: `¯\_(ツ)_/¯` not in any recent (i.e., in any of the changes I made). Didn't touch this sort of thing.

**Comment by rvantonder** [comment]
> It seems the code which generates these suggestions is quite different. This PR you can probably merge, and we need to do the escaping in the frontend layer?

Hmm, not sure. I'm not trying to create work here that fixes search logic  (I didn't investigate what this actually does or is supposed to do). I'm just trying to get rid of stuff that's being confusingly used and distracting me from understanding things.

That said, I don't really want to pick up ownership of fixing this right now, so if this is something that matters and should be used then maybe a follow up to #7031 makes sense since that's a recent change. LMK which direction you prefer to go (merge this and create an issue, or close this and fix soon, or something else).

**Comment by keegancsmith** [comment]
Ok I agree, this should be merged in. I have re-opened https://github.com/sourcegraph/sourcegraph/issues/6498

**Pull request merged**